cirq-google/cirq_google/engine/engine_job.py

```diff
@@ -299,12 +299,15 @@ async def _await_result_async(self) -> quantum.QuantumResult:
                     'Internal error: The job response type is not recognized.'
                 )  # pragma: no cover
 
+        # The desired rate limit is 1000 QPM which is ~15 QPS
+        limiter = duet.limiter(15)
         async with duet.timeout_scope(self.context.timeout):  # type: ignore[arg-type]
             while True:
+                slot = await limiter.acquire()  # Enforce rate limit
```

Comment:
This doesn't appear to have an effect across multiple jobs since `limiter` is scoped to a single invocation of `_await_result_async()`.